cirq-core/cirq/transformers/gauge_compiling/gauge_compiling_test_utils.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from unittest.mock import patch
+import pytest
+
+import numpy as np
+
+import cirq
+from cirq.transformers.gauge_compiling import GaugeTransformer, GaugeSelector
+
+
+class GaugeTester:
+
+    two_qubit_gate: cirq.Gate
+    gauge_transformer: GaugeTransformer
+
+    @pytest.mark.parametrize('generation_seed', [*range(5)])
+    @pytest.mark.parametrize('transformation_seed', [*range(5)])
```

Comment:
The 2 parametrize decorators do Cartesian product so there is 25 test cases - is that necessary?
Perhaps we can combine to a single parametrization of 2 seed values.
Also consider using `np.random.randint(2**31, size=(5, 2)).tolist()` for the seed values in parametrization to check with non-repetitive seeding.

Reply:
done, but instead of using `np.random.randint` I used `np.random.RandomState(0).randint` inorder to make the test reproducible